cmake/select_tdi_target.cmake

```diff
@@ -0,0 +1,109 @@
+# Select TDI target type.
+#
+# Copyright 2022-2023 Intel Corporation
+# SPDX-License-Identifier: Apache 2.0
+#
+# Processes the new TDI_TARGET string option and the legacy Boolean target
+# options (DPDK_TARGET, ES2K_TARGET, and TOFINO_TARGET).
+#
+# Issues deprecation messages for legacy target options that are enabled.
+#
+# Sets TARGETFLAG and the corresponding Boolean target option.
+# Unsets all other Boolean target options.
+#
+
+set(TDI_TARGET "" CACHE STRING "TDI Target Type")
+
+set(_valid_tdi_targets DPDK ES2K TOFINO)
```

Comment:
Is this where we would add more targets if supported in future?

Reply:
If it is a simple addition here, thats good. If more places need manual editing, will need to document (or add a comment here) that other changes may be needed elsewhere

Reply:
> Is this where we would add more targets if supported in future?

Yes, that's the plan.  If it turns out we need the information elsewhere (which I don't expect to happen), we can rename the symbol to `VALID_TDI_TARGETS` and remove the command that unsets it at the end of the module.